src/pal/src/CMakeLists.txt

```diff
@@ -275,6 +275,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
 endif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
 
 if(CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
+  # some  dependencies from port collection are installed under /usr/local
+  link_directories("/usr/local/lib")
```

Comment:
I was thinking about it a bit and I would actually prefer a different solution (I've verified that it works). Instead of adding the /usr/local/lib to the library directories, could you please modify the src/scripts/genLttngProvider.py as follows? We should be using the find_library for all non-system libraries.
```diff
diff --git a/src/scripts/genLttngProvider.py b/src/scripts/genLttngProvider.py
index de537d0f1a..296042be5b 100644
--- a/src/scripts/genLttngProvider.py
+++ b/src/scripts/genLttngProvider.py
@@ -555,8 +555,10 @@ def generateLttngFiles(etwmanifest,eventprovider_directory):

         tracepointprovider_Cmake.write("""    )

+        find_library(LTTNG NAMES lttng-ust)
+
         target_link_libraries(coreclrtraceptprovider
-                              -llttng-ust
+                              ${LTTNG}
         )

         # Install the static coreclrtraceptprovider library

```

Reply:
I updated code as you suggested @janvorli (and verified it works for me) 

It is interesting that we already add /usr/local/include to include path but not lib/lib64 for libraries.